File: apps/commerce-api/src/main/java/com/loopers/domain/product/ProductCommand.java
package com.loopers.domain.product;

import java.util.List;

public class ProductCommand {
    public record GetAvailable(
        List<OrderOption> options
    ) {
    }

    public record OrderOption(
        Long productId, Long productOptionId, Integer quantity
    ) {
    }

    public record Search(
        SortBy sortBy,
        int page,
        int size,
        ProductStatus status
    ) {
    }
}

File: apps/commerce-streamer/src/main/kotlin/com/loopers/domain/product/ProductCommand.kt
package com.loopers.domain.product

class ProductCommand {
    data class Deduct(
        val options: List<Option>
    ) {
        fun productIds(): List<Long> {
            return options.map { it.productId }.toList()
        }

        fun productOptionIds(): List<Long> {
            return options.map { it.productOptionId }.toList()
        }
    }

    data class Option(
        val productId: Long,
        val productOptionId: Long,
        val quantity: Int
    )
}
